pkg/arvo/app/graph-store.hoon

```diff
@@ -311,6 +315,37 @@
         (add-node-list resource graph mark (sort-nodes nodes))
       ==
       ::
+      ++  check-for-duplicates
+        |=  [=resource:store nodes=(set index:store)]
+        ^-  ?
+        =|  has-duplicates=_|
+        =/  node-list  ~(tap in nodes)
+        |-
+        ?~  node-list
+          has-duplicates
+        =?  has-duplicates  (has-node resource i.node-list)  %.y
```

Comment:
Isn't this equivalent to:

```suggestion
        =.  has-duplicates  |(has-duplicates (has-node resource i.node-list))
```

If yes, why do we try the entire node-list, instead of returning `%.y` as soon as we find a duplicate?

Reply:
Also a good point. Addressed this also.